flytekit/core/condition.py

```diff
@@ -183,11 +183,11 @@ def end_branch(self) -> Union[Condition, Promise]:
             if self._selected_case.output_promise is None and self._selected_case.err is None:
                 raise AssertionError("Bad conditional statements, did not resolve in a promise")
             elif self._selected_case.output_promise is not None:
-                return self._compute_outputs(self._selected_case.output_promise)
+                return typing.cast(Promise, self._compute_outputs(self._selected_case.output_promise))
             raise ValueError(self._selected_case.err)
         return self._condition
 
-    def _compute_outputs(self, selected_output_promise) -> Union[Promise, Tuple[Promise], VoidPromise]:
+    def _compute_outputs(self, selected_output_promise) -> Optional[Union[Tuple[Promise], Promise, VoidPromise]]:
```

Comment:
I assume the `Tuple[Promise]` is coming from `create_task_output` - is that correct?  it looks kinda weird.

Reply:
yes, `create_task_output` will return `Optional[Union[Tuple[Promise], Promise]]` https://github.com/flyteorg/flytekit/blob/28153cc787bdd529f9f6050c393687715476153d/flytekit/core/promise.py#L470-L472